dev/lint-python

```diff
@@ -82,6 +87,34 @@ else
     rm "$PYCODESTYLE_REPORT_PATH"
 fi
 
+# Check python document style, skip check if pydocstyle is not installed.
+if hash "$PYDOCSTYLEBUILD" 2> /dev/null; then
+    if [ $PYDOCSTYLEVERSION == "2.1.1" ]; then
+        pydocstyle --config=dev/tox.ini $DOC_PATHS_TO_CHECK >> "$PYDOCSTYLE_REPORT_PATH"
+        pydocstyle_status="${PIPESTATUS[0]}"
+
+        if [ "$compile_status" -eq 0 -a "$pydocstyle_status" -eq 0 ]; then
+            lint_status=0
+        else
+            lint_status=1
+        fi
+
+        if [ "$lint_status" -ne 0 ]; then
```

Comment:
could you just move this block above and get rid of `lint_status`?

Reply:
not sure i understand

Reply:
So if I understand @BryanCutler's request correctly, instead of setting linte_status in the if check above and then `lint_status` to determine if we should run the pydocstyle checker, just directly check the compile status and then run the pydocstyle checker on that.

That being said I think we could also drop the compile_status check since we would have already exited the script if it was non-zero at this point (I think).